core/src/main/java/com/klaviyo/coresdk/KlaviyoConfig.kt

```diff
@@ -0,0 +1,34 @@
+package com.klaviyo.coresdk
+
+data class KlaviyoConfig (
```

Comment:
Originally when speccing out the project I felt this was the way to go for allowing the user to define the configuration of our SDK. Now I'm feeling less certain that this is the case. The alternative is to apply configuration in some supporting files, perhaps XML config, or to create a Kotlin object class (basically a singleton with public vars we can write to) to store this informtion.

Reply:
I thought about turning this into a singleton but it feels like an anti-pattern to be doing that in a builder

Reply:
i think this is fine, in terms of getting the apikey, its easier to add a variable to the build config thats backed by an env var and do BuildConfig.KLAVIYO_API_KEY than it is to add a metadata property to the xml

Reply:
this pattern is fine but i think we need to consider blocking out the KlaviyoConfig constructor. do we want to force devs to use the builder?

Reply:
also i think the data keyword isn't necessary since we dont need the benefits of data like `.copy` and autogenerated comparison functions

Reply:
Those are fair points. Since technically this class exists to store data, I blindly attached the data keyword, but there's not going to be any need for comparisons or string generation in this class so its definitely unnecessary overhead.

I do think its best to force the users to use the Builder so I will try to block out the constructor.

Reply:
Also thanks this gives me a bit more confidence in this approach again